src/shared/components/CSVExportButton.tsx

```diff
@@ -1,72 +1,102 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import {connect, ConnectedProps} from 'react-redux'
 
 // Components
-import {Button, ComponentStatus, IconFont} from '@influxdata/clockface'
-
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {
+  Button,
+  ComponentStatus,
+  ComponentColor,
+  IconFont,
+} from '@influxdata/clockface'
+
+// Selectors and Actions
+import {getActiveQuery} from 'src/timeMachine/selectors'
+import {
+  runDownloadQuery,
+  DOWNLOAD_EVENT_COMPLETE,
+} from 'src/timeMachine/actions/queries'
 
 // Types
-import {AppState} from 'src/types'
+import {AppState, RemoteDataState} from 'src/types'
+
+type Props = ConnectedProps<typeof connector>
 
-interface StateProps {
-  files: string[] | null
+interface State {
+  state: RemoteDataState
 }
 
-class CSVExportButton extends PureComponent<StateProps, {}> {
+class CSVExportButton extends PureComponent<Props, State> {
+  private controller: AbortController
+
+  constructor(props) {
+    super(props)
+    this.controller = new AbortController()
+    this.controller.signal.addEventListener(DOWNLOAD_EVENT_COMPLETE, () => {
+      this.setState({state: RemoteDataState.Done})
+    })
+    this.state = {state: RemoteDataState.NotStarted}
```

Comment:
Why doesn't the linter complain when we set state directly here?

Reply:
it's in the constructor. That is permitted.